### DISCUSSION TITLE
Importing a random field into Moose

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#16935] [Q&A General] [answered]
Hi all,

I have a file external to moose that contains stochastic field data for the Young's Modulus. The file contains data for each node on the grid, as well as XY coordinates for each node. Additionally, I have the .msh file used to create this field. 
The domain represents a hollow quarter circle. 
I like to import this data into moose and then impose it onto a more refined mesh than one used to generate the field. 
My first thought was to use the function PiecewiseMultilinear, that but that seems to only support square domains. 

If possible, what would be the best way to import this data into moose so that I can utilize the random field for an FEA simulation?

**Comment by YaqiWang** [comment, marked as answer]
Are you able to convert this data file into an exodus file? If so, you can use `SolutionUserObject` to load the data and create a function in type of `SolutionFunction` from it. You might not have to use exodus. I am not sure if gmsh has a format for storing nodal solutions and if libMesh can load it. But `SolutionFunction` sounds to me the must.

**Reply by TheGreatCid** [reply]
To my knowledge, I don't have a way to convert to exodus as my work has been done in MATLAB. I was hoping to use moose to convert to exodus. I do not believe gmsh can store nodal data. 
Would you have a suggestion for how to convert to exodus?

**Reply by YaqiWang** [reply]
Maybe Python has some nice interface to exodus to allow us create a file from msh file. Or you can let moose to load msh file and create a nodal aux kernel to load the data and assign them to nodes?

**Reply by TheGreatCid** [reply]
Thanks for your replies

I trying to create a nodal aux kernel may be the best shot. I will give that a try .
Thanks for the advice
